torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -271,6 +287,43 @@ static void UpdateScalarTypeForOutput(
       CreateProfiledTensorTypeWithScalarType(output_tensor_type, scalar_type));
 }
 
+static void RecoverScalarTypeForOutput(
+    Value* out,
+    const c10::ScalarType& scalar_type) {
+  Node* n = out->node();
+  TORCH_INTERNAL_ASSERT(nullptr != n);
+  const int64_t onnx_type = ScalarTypeToONNXType(scalar_type);
+  Node* cast_node = n->owningGraph()->create(onnx::Cast, 1);
+  cast_node->addInput(out);
+  cast_node->i_(attr::to, onnx_type);
+  cast_node->insertAfter(n);
+  out->replaceAllUsesAfterNodeWith(cast_node, cast_node->output());
+}
+
+static void LowPrecisionCastNodeForStandardOps(Node* n, int opset_version) {
+  TORCH_INTERNAL_ASSERT(n->outputs().size() == 1);
+  TORCH_INTERNAL_ASSERT(n->output()->type()->cast<TensorType>() != nullptr);
```

Comment:
Please check when to use assertion, and when to use controlflow, for nullptr checking.

Reply:
Do you mean that we should use controlflow to return the function when n->output()->type()->cast<TensorType>() equal to nullptr. Or it's reasonable for nullptr and the program should continue to run?